src/Resources.pkl

```diff
@@ -0,0 +1,97 @@
+module concourse_pipeline.Resources
+
+import "./Pipeline.pkl"
+
+open class GitResource extends Pipeline.Resource{
+  type = "git"
+  icon = "github"
+  source: GitResourceConfig?
+}
+
+class GitResourceConfig{
+  uri: Uri
+  branch: String?
+  private_key: String?
+  private_key_user: String?
+  private_key_passphrase: String?
+  forward_agent: Boolean?
+  username: String?
```

Comment:
(minor) Could make username and password fields mandatory too so we don't end up getting rate limited when resource is used without these fields

Reply:
I think we want the classes here to reflect the resource type docs. We can then we can exend for our internal usage.